frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ImportCluster/ImportCluster.test.tsx

```diff
@@ -919,29 +1069,73 @@ describe('Import Discovered Cluster', () => {
 
     await waitForNocks([projectNock, managedClusterNock, kacNock, importCommandNock, autoImportSecretNock])
   })
+})
 
-  test('sets discovered OCP cluster URL field', async () => {
-    const { getAllByText, getAllByLabelText, getByDisplayValue } = render(<Component />) // Render component
+describe('Import Discovered Cluster with import credentials', () => {
+  beforeEach(() => {
+    localStorage.clear()
+    nockIgnoreRBAC()
+    nockIgnoreApiPaths()
+    nockIgnoreOperatorCheck()
+  })
+  test('create discovered ROSA cluster with auto import service account', async () => {
+    window.sessionStorage.setItem('DiscoveredClusterConsoleURL', 'https://test-cluster-serviceaccount.com')
 
-    await waitFor(() => expect(getAllByText(mockDiscoveredClusters[0].metadata.name!)[0]!).toBeInTheDocument()) // Wait for Discovered ROSA Cluster to appear in table
-    userEvent.click(getAllByLabelText('Actions')[0]) // Click on Kebab menu
+    const Component = () => (
+      <RecoilRoot
+        initializeState={(snapshot) => {
+          snapshot.set(managedClusterSetsState, [mockManagedClusterSet])
+          snapshot.set(secretsState, [mockCRHCredential2, mockOCMConnection2])
+          snapshot.set(discoveryConfigState, [mockDiscoveryConfig])
+          snapshot.set(discoveredClusterState, mockDiscoveredClusters)
+          snapshot.set(namespacesState, mockNamepaces)
+        }}
+      >
+        <MemoryRouter>
+          <Routes>
+            <Route path="/" element={<DiscoveredClustersPage />} />
+            <Route path={NavigationPath.importCluster} element={<ImportClusterPage />} />
+          </Routes>
+        </MemoryRouter>
+      </RecoilRoot>
+    )
+
+    const { getAllByText, getAllByLabelText, getByDisplayValue } = render(<Component />) // Render the custom component
 
+    await waitFor(() => {
+      expect(getAllByText(mockDiscoveredClusters[2].metadata.name!)[0]!).toBeInTheDocument()
+    })
+    userEvent.click(getAllByLabelText('Actions')[1]) // Click on Kebab menu
```

Comment:
I inserted a `screen.logTestingPlaygroundURL()` just before this line to debug. From the link, I could see the list of discovered clusters. The cluster you added is the 2nd one called "rosa-discovery-cluster", so you want index 2 instead of 1 out of all the kebab menus labeled "Actions".

![image](https://github.com/user-attachments/assets/472af100-665d-4d63-8dca-05912416699b)

That will get you to the next issue, which is with the nocking.